agent/agent.go

```diff
@@ -77,7 +78,13 @@ var (
 	sessionCookieCacheLimit = flag.Int("session-cookie-cache-limit", 1000, "Upper bound on the number of concurrent sessions that can be tracked by the agent")
 	rewriteWebsocketHost    = flag.Bool("rewrite-websocket-host", false, "Whether to rewrite the Host header to the original request when shimming a websocket connection")
 
+	projectID		= flag.String("project-id", "", "Name of the GCP project id")
```

Comment:
Instead of adding flags for the project ID, zone, and instance, let's have the agent code read them from the metadata server (if one is present).

Reply:
Actually, for project ID itself, it seems to  required by the monitoring API, so let's keep it but rename it to something like `monitoring-project-id` to make it clear *which* project needs to be provided here.

For the instance and zone, let's instead make the flags for the monitored resource more generic and add a type (so that we aren't assuming `gce_instance`), and a comma-separated list of key=value pairs for the resource labels.

That way we don't have to rely on the metadata server or limit ourselves to only monitoring GCE VMs.

Reply:
For this, I removed the instanceID and instanceZone flags and added flags for resourceType and resourceLabels (that expects key=value pairs) to support monitoring resources other than `gce_instance`.